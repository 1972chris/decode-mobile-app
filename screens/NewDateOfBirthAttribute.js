import React from 'react';
import { SecureStore } from 'expo';
import { connect } from 'react-redux';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import DatePicker from 'react-native-datepicker';
import Button from '../application/components/Button/Button';
import styles from './styles';
import { saveDateOfBirth } from '../application/redux/actions/attributes';

const maxDate = new Date();
const minDate = new Date();
minDate.setFullYear(minDate.getFullYear() - 130);

class NewDateOfBirthAttribute extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      dateOfBirth: '',
    };
  }


  render() {
    return (
      <View style={styles.attributesManagementContainer}>
        <DatePicker
          style={{ width: 200 }}
          date={this.state.dateOfBirth}
          mode="date"
          placeholder="select date"
          format="DD/MM/YYYY"
          minDate={this.props.minDate}
          maxDate={this.props.maxDate}
          confirmBtnText="Confirm"
          cancelBtnText="Cancel"
          customStyles={{
            dateInput: {
              marginLeft: 36,
            },
          }}
          onDateChange={(date) => {
            this.setState({ dateOfBirth: date });
          }}
        />
        <Button
          name="addDateOfBirth"
          onPress={() => this.props.saveDateOfBirth(this.state.dateOfBirth, this.props.walletId)}
        />
      </View>
    );
  }
}


NewDateOfBirthAttribute.propTypes = {
  saveDateOfBirth: PropTypes.func.isRequired,
  walletId: PropTypes.string.isRequired,
  minDate: PropTypes.instanceOf(Date),
  maxDate: PropTypes.instanceOf(Date),
};

NewDateOfBirthAttribute.defaultProps = {
  minDate,
  maxDate,
};

const mapStateToProps = state => ({
  walletId: state.wallet.id,
});

const mapDispatchToProps = dispatch => ({
  saveDateOfBirth: (dateOfBirth, walletId) =>
    dispatch(saveDateOfBirth(dateOfBirth, walletId, SecureStore.setItemAsync)),
});

export default connect(mapStateToProps, mapDispatchToProps)(NewDateOfBirthAttribute);
