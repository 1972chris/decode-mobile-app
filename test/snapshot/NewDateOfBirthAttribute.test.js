import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import NewDateOfBirthAttribute from '../../screens/NewDateOfBirthAttribute';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore([thunk]);

const initialState = {
  navigation: {},
  wallet: {
    id: '123',
  },
};

it('renders NewDateOfBirthAttribute component', () => {
  const wrapper = shallow(
    <NewDateOfBirthAttribute
      minDate={new Date(Date.UTC(1950, 1, 1))}
      maxDate={new Date(Date.UTC(2019, 1, 1))}
    />,
    { context: { store: mockStore(initialState) } },
  );
  expect(wrapper.dive()).toMatchSnapshot();
});
