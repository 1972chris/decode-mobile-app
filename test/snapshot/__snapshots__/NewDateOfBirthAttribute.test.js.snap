// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NewDateOfBirthAttribute component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewDateOfBirthAttribute
    maxDate={2018-12-31T23:00:00.000Z}
    minDate={1949-12-31T23:00:00.000Z}
    saveDateOfBirth={[Function]}
    walletId="123"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <DatePicker
          TouchableComponent={[Function]}
          allowFontScaling={true}
          androidMode="default"
          cancelBtnText="Cancel"
          confirmBtnText="Confirm"
          customStyles={
            Object {
              "dateInput": Object {
                "marginLeft": 36,
              },
            }
          }
          date=""
          disabled={false}
          duration={300}
          format="DD/MM/YYYY"
          height={259}
          hideText={false}
          iconSource={1}
          maxDate={2018-12-31T23:00:00.000Z}
          minDate={1949-12-31T23:00:00.000Z}
          modalOnResponderTerminationRequest={[Function]}
          mode="date"
          onDateChange={[Function]}
          placeholder="select date"
          showIcon={true}
          style={
            Object {
              "width": 200,
            }
          }
        />,
        <Button
          enabled={true}
          name="addDateOfBirth"
          onPress={[Function]}
          style={Object {}}
        />,
      ],
      "style": Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFF",
        "margin": 16,
        "marginTop": 30,
        "paddingHorizontal": 20,
        "paddingVertical": 40,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "TouchableComponent": [Function],
          "allowFontScaling": true,
          "androidMode": "default",
          "cancelBtnText": "Cancel",
          "confirmBtnText": "Confirm",
          "customStyles": Object {
            "dateInput": Object {
              "marginLeft": 36,
            },
          },
          "date": "",
          "disabled": false,
          "duration": 300,
          "format": "DD/MM/YYYY",
          "height": 259,
          "hideText": false,
          "iconSource": 1,
          "maxDate": 2018-12-31T23:00:00.000Z,
          "minDate": 1949-12-31T23:00:00.000Z,
          "modalOnResponderTerminationRequest": [Function],
          "mode": "date",
          "onDateChange": [Function],
          "placeholder": "select date",
          "showIcon": true,
          "style": Object {
            "width": 200,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "enabled": true,
          "name": "addDateOfBirth",
          "onPress": [Function],
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <DatePicker
            TouchableComponent={[Function]}
            allowFontScaling={true}
            androidMode="default"
            cancelBtnText="Cancel"
            confirmBtnText="Confirm"
            customStyles={
              Object {
                "dateInput": Object {
                  "marginLeft": 36,
                },
              }
            }
            date=""
            disabled={false}
            duration={300}
            format="DD/MM/YYYY"
            height={259}
            hideText={false}
            iconSource={1}
            maxDate={2018-12-31T23:00:00.000Z}
            minDate={1949-12-31T23:00:00.000Z}
            modalOnResponderTerminationRequest={[Function]}
            mode="date"
            onDateChange={[Function]}
            placeholder="select date"
            showIcon={true}
            style={
              Object {
                "width": 200,
              }
            }
          />,
          <Button
            enabled={true}
            name="addDateOfBirth"
            onPress={[Function]}
            style={Object {}}
          />,
        ],
        "style": Object {
          "alignSelf": "stretch",
          "backgroundColor": "#FFF",
          "margin": 16,
          "marginTop": 30,
          "paddingHorizontal": 20,
          "paddingVertical": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "TouchableComponent": [Function],
            "allowFontScaling": true,
            "androidMode": "default",
            "cancelBtnText": "Cancel",
            "confirmBtnText": "Confirm",
            "customStyles": Object {
              "dateInput": Object {
                "marginLeft": 36,
              },
            },
            "date": "",
            "disabled": false,
            "duration": 300,
            "format": "DD/MM/YYYY",
            "height": 259,
            "hideText": false,
            "iconSource": 1,
            "maxDate": 2018-12-31T23:00:00.000Z,
            "minDate": 1949-12-31T23:00:00.000Z,
            "modalOnResponderTerminationRequest": [Function],
            "mode": "date",
            "onDateChange": [Function],
            "placeholder": "select date",
            "showIcon": true,
            "style": Object {
              "width": 200,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "enabled": true,
            "name": "addDateOfBirth",
            "onPress": [Function],
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
